### PR TITLE
IP/LEC2_Q4

### PR DESCRIPTION
**Pull request by htshllwn**
Given two strings, check if they are permutations of each other. Return true or false.
Permutation means - length of both the strings should same and should contain same set of characters. Order of characters doesn't matter.
Note : Input strings contain only lowercase english alphabets.
**Input format :**
```
Line 1 : String 1
Line 2 : String 2
```
**Sample Input 1 :**
```
abcde
baedc
```
**Sample Output 1 :**
`true`
**Sample Input 2 :**
```
abc
cbd
```
**Sample Output 2 :**
`false`

### CODE REVIEW

File: Interview Preparation 1Mar2019/Lecture 2 - Arrays & Strings/Lecture/Q4 - Check Permutation/Main.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        for (int i = 0; i < counts1.length; i++) {
+            if (counts1[i] != counts2[i])
+                return false;
+        }
+        return true;
+    }
+
+    public static int[] countsArray(String str) {
+        int lb = 97;
+        int[] counts = new int[26];
```
Comment (dgr8akki) [inline Interview Preparation 1Mar2019/Lecture 2 - Arrays & Strings/Lecture/Q4 - Check Permutation/Main.java:53]: Will it work if my strings are
```This is testing```
```Testing is this```

File: Interview Preparation 1Mar2019/Lecture 2 - Arrays & Strings/Lecture/Q4 - Check Permutation/Main.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        int[] counts2 = countsArray(input2);
+        for (int i = 0; i < counts1.length; i++) {
+            if (counts1[i] != counts2[i])
+                return false;
+        }
+        return true;
+    }
+
+    public static int[] countsArray(String str) {
+        int lb = 97;
```
Comment (dgr8akki) [inline Interview Preparation 1Mar2019/Lecture 2 - Arrays & Strings/Lecture/Q4 - Check Permutation/Main.java]: Selecting lower bound as 97 which is ```a``` will break if my string contains an upper case character.

File: Interview Preparation 1Mar2019/Lecture 2 - Arrays & Strings/Lecture/Q4 - Check Permutation/Main.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        int[] counts = new int[26];
+        for (int i = 0; i < str.length(); i++) {
+            char c = str.charAt(i);
+            int ascii = (int) c;
+            int index = ascii - lb;
+            counts[index] += 1;
```
Comment (dgr8akki) [inline Interview Preparation 1Mar2019/Lecture 2 - Arrays & Strings/Lecture/Q4 - Check Permutation/Main.java]: You can use shorthand here, like
```
for (int i = 0; i < str.length(); i++)
  counts[str.charAt(i) - 'a']++;
```

File: Interview Preparation 1Mar2019/Lecture 2 - Arrays & Strings/Lecture/Q4 - Check Permutation/Main.java (hunk 1 above)
Comment (htshllwn) [reply Interview Preparation 1Mar2019/Lecture 2 - Arrays & Strings/Lecture/Q4 - Check Permutation/Main.java:53]: No it wont work
Check the Note in the question
**Note : Input strings contain only lowercase english alphabets.**

File: Interview Preparation 1Mar2019/Lecture 2 - Arrays & Strings/Lecture/Q4 - Check Permutation/Main.java (hunk 2 above)
Comment (htshllwn) [reply Interview Preparation 1Mar2019/Lecture 2 - Arrays & Strings/Lecture/Q4 - Check Permutation/Main.java]: Input will never contain an uppercase character. Check Note.

File: Interview Preparation 1Mar2019/Lecture 2 - Arrays & Strings/Lecture/Q4 - Check Permutation/Main.java (hunk 3 above)
Comment (htshllwn) [reply Interview Preparation 1Mar2019/Lecture 2 - Arrays & Strings/Lecture/Q4 - Check Permutation/Main.java]: This looks much cleaner. Wasn't aware of this in JAVA. Changing in next commit.

**Review by dgr8akki — APPROVED** [review]

**Review by shradheyt — APPROVED** [review]

**Pull request merged**